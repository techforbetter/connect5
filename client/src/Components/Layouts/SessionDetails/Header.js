import React from "react";
import {
  Header,
  Heading,
}
  from "./styledComponents";

const SessionHeader = () => (
  <Header>
    <Heading>
      <h1>Session Details</h1>
    </Heading>
  </Header>
);

export default SessionHeader;
