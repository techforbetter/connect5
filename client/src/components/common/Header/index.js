import React from 'react';
import { connect } from 'react-redux';
import styled, { css } from 'styled-components';
import { NavLink } from 'react-router-dom';

import { colors, borders, shadows } from '../../../theme';

import Connect5Logo from '../../../assets/connect-5-white.png';

import { HOME_URL } from '../../../constants/navigationRoutes';
import { checkBrowserWidth } from '../../../actions/checkBrowserWidth';
import HumburgerMenu from '../Menu';

const sharedStyles = css`
  position: fixed;
  left: 0;
  width: 100%;
  box-shadow: ${shadows.primary};
  color: ${colors.white};
  padding: 0.5rem;
  z-index: 1000;
`;

export const sectionHeader = css`
  display: flex;
  justify-content: flex-start;
  top: ${props => (props.nudge ? '31px' : 0)};
  background-color: ${colors.primary};
  color: ${colors.offWhite};
  height: 48px;
  text-align: left;
  text-transform: capitalize;
  h1 {
    font-size: 1.3rem;
    font-weight: 400;
    position: relative;
    color: ${colors.offWhite};
    &::after {
      content: ' ';
      position: absolute;
      border-bottom: ${borders.header};
      bottom: -12px;
      left: 0;
      width: 100%;
    }
  }
`;

export const viewHeader = css`
  top: 0;
  background-color: ${colors.black};
  color: ${colors.offWhite};
  height: 32px;
  text-align: center;
  text-transform: capitalize;
  h1 {
    font-size: 1rem;
    font-weight: 300;
    color: ${colors.offWhite};
  }
`;

export const homeHeader = css`
  top: 0;
  background-color: ${colors.primary};
  color: ${colors.offWhite};
  height: 48px;
  display: flex;
  justify-content: space-between;
`;

const StyledHeader = styled.div`
  ${sharedStyles};
  ${props => props.type === 'section' && sectionHeader};
  ${props => props.type === 'view' && viewHeader};
  ${props => props.type === 'home' && homeHeader};
`;

const HomeIcon = styled(NavLink)`
  text-decoration: none;
  color: ${colors.offWhite};
  i {
    font-size: 30px;
  }
`;

const Logo = styled.img`
  height: 32px;
  margin: 0;
`;

// NOTE: If you need Section and View header at same time
// then give your section header the nudge prop to shift it down

const Header = ({
  label,
  type,
  userRole,
  isDeskTop,
  isMobile,
  isAuthenticated,
  ...props
}) => {
  return (
    <>
      {type === 'home' ? (
        <StyledHeader type={type} {...props}>
          {isAuthenticated ? (
            <HumburgerMenu />
          ) : (
            <Logo src={Connect5Logo} alt="logo" />
          )}
          <HomeIcon to={userRole ? '/participant-dashboard' : HOME_URL}>
            <i className="fas fa-home" />
          </HomeIcon>
        </StyledHeader>
      ) : (
        <StyledHeader type={type} {...props}>
          {isAuthenticated && type === 'section' && <HumburgerMenu />}
          <h1>{label}</h1>
        </StyledHeader>
      )}
    </>
  );
};

const mapStateToProps = state => ({
  width: state.checkBrowserWidth.width,
  isMobile: state.checkBrowserWidth.isMobile,
  isDeskTop: state.checkBrowserWidth.isDeskTop,
  isAuthenticated: state.auth.isAuthenticated,
});

export default connect(
  mapStateToProps,
  { checkBrowserWidth }
)(Header);
