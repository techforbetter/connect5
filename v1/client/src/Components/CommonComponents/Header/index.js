import React, { Component } from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";
import Logout from "../Logout/index";
import Logo from "../../../assets/connect5_logo_main.jpg";

const Container = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 55px;
  box-sizing: border-box;
  padding: 0 20px;
  background-color: #fff;
  z-index: 10000;

  + * {
    margin-top: 55px;
  }

  :after {
    content: "";
    display: block;
    width: 100%;
    background-color: var(--light-gray-border);
    position: absolute;
    top: 55px;
    height: 3px;
    left: 0;
  }
`;


const I = styled.i`
  font-size: 35px;
  color: var(--home-color);
  margin-right: 6px;
`;

const Icons = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-left: 75px;
  height: 100%
`;

const Image = styled.img`
  height: 50px;
`;


class Header extends Component {
  state= {
    login: false,
  }

  componentDidMount() {
    if (localStorage.jwtToken) {
      this.setState({ login: true });
    }
  }

  logout = () => {
    localStorage.removeItem("jwtToken");
    this.setState({ login: false });
    window.location.href = "/";
  }

  render() {
    const { logout } = this;
    const { login } = this.state;
    return (
      <Container>
        <Image src={Logo} alt="Logo" />
        <Icons>
          <Link to="/">
            <I className="fas fa-home" />
          </Link>
          {login ? <Logout logout={logout} /> : null}
        </Icons>
      </Container>

    );
  }
}

export default Header;
