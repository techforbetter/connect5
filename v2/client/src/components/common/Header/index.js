import React from 'react';
import styled, { css } from 'styled-components';
import { NavLink } from 'react-router-dom';

import { colors, borders, shadows } from '../../../theme';

import Connect5Logo from '../../../assets/connect-5-white.png';

import { HOME_URL } from '../../../constants/navigationRoutes';

const sharedStyles = css`
  position: fixed;
  left: 0;
  width: 100%;
  box-shadow: ${shadows.primary};
  color: ${colors.white};
  padding: 0.5rem;
  z-index: 1000;
`;

export const sectionHeader = css`
  top: ${props => (props.nudge ? '31px' : 0)};
  background-color: ${colors.primary};
  color: ${colors.offWhite};
  height: 48px;
  text-align: left;
  text-transform: capitalize;

  h1 {
    font-size: 1.5rem;
    font-weight: 400;
    position: relative;
    color: ${colors.offWhite};

    &::after {
      content: ' ';
      position: absolute;
      border-bottom: ${borders.header};
      bottom: -2px;
      left: 0;
      width: 50%;
      @media (min-width: 768px) {
        width: 25%;
    }
  }
`;

export const viewHeader = css`
  top: 0;
  background-color: ${colors.black};
  color: ${colors.offWhite};
  height: 32px;
  text-align: center;
  text-transform: capitalize;

  h1 {
    font-size: 1rem;
    font-weight: 300;
    color: ${colors.offWhite};
  }
`;

export const homeHeader = css`
  top: 0;
  background-color: ${colors.primary};
  color: ${colors.offWhite};
  height: 48px;
  display: flex;
  justify-content: space-between;
`;

const StyledHeader = styled.div`
  ${sharedStyles}
  ${props => props.type === 'section' && sectionHeader}
  ${props => props.type === 'view' && viewHeader}
  ${props => props.type === 'home' && homeHeader}
`;

const HomeIcon = styled(NavLink)`
  text-decoration: none;
  color: ${colors.offWhite};
  i {
    font-size: 30px;
  }
`;

const Logo = styled.img`
  height: 32px;
  margin: 0;
`;

// NOTE: If you need Section and View header at same time
// then give your section header the nudge prop to shift it down

const Header = ({ label, type, ...props }) => {
  return (
    <>
      {type === 'home' ? (
        <StyledHeader type={type} {...props}>
          <HomeIcon to={HOME_URL}>
            <i className="fas fa-home" />
          </HomeIcon>
          <Logo src={Connect5Logo} alt="logo" />
        </StyledHeader>
      ) : (
        <StyledHeader type={type} {...props}>
          <h1>{label}</h1>
        </StyledHeader>
      )}
    </>
  );
};

export default Header;
