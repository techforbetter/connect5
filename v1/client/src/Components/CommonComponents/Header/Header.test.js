import React from "react";
import renderer from "react-test-renderer";
import { StaticRouter } from "react-router-dom";

import Header from "./index";

test("Header matches snapshot", () => {
  const context = {};
  const component = renderer.create(
    <StaticRouter context={context}>
      <Header />
    </StaticRouter>,
  );
  expect(component).toMatchSnapshot();
});
