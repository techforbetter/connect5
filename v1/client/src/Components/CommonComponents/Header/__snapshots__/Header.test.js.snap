// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div
  className="sc-bxivhb igcOnr"
>
  <img
    alt="Logo"
    className="sc-bZQynM fSermK"
    src="connect5_logo_main.jpg"
  />
  <div
    className="sc-EHOje eLNQHg"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <i
        className="fas fa-home sc-ifAKCX bmGznW"
      />
    </a>
  </div>
</div>
`;
